lib/generator.js

```diff
@@ -340,56 +338,67 @@ class Generator {
-            installedPkg = require(path.resolve(DEFAULT_TEMPLATES_DIR, pkg.name, PACKAGE_JSON_FILENAME));
-          } else { // Template is not a filesystem path...
-            const templatePath = path.resolve(DEFAULT_TEMPLATES_DIR, this.templateName);
-            if (await isLocalTemplate(templatePath)) {
-              // This "if" is covering the following workflow:
-              // ag asyncapi.yaml ../html-template
-              // The previous command installs a template called @asyncapi/html-template
-              // And now we run the command again but with the resolved name:
-              // ag asyncapi.yaml @asyncapi/html-template
-              // The template name doesn't look like a file system path but we find
-              // that the package is already installed and it's a symbolic link.
-              const { resolvedLink } = await getLocalTemplateDetails(templatePath);
-              log.debug(`This template has already been installed and it's pointing to your filesystem at ${resolvedLink}.`);
-            }
-            installedPkg = require(path.resolve(templatePath, PACKAGE_JSON_FILENAME));
+            pkgPath = path.resolve(this.templateName);
+            installedPkg = require(path.resolve(pkgPath, PACKAGE_JSON_FILENAME));
+            
+            log.debug('Remember that your local template must have its own node_modules, "npm install" is not triggered.');
+          } else {
+            pkgPath = resolvePkg(this.templateName);
+            installedPkg = require(path.join(pkgPath, PACKAGE_JSON_FILENAME));
           }
+          packageVersion = installedPkg.version;
+          log.debug(`Template sources taken from ${pkgPath}.`);
+          if (packageVersion) log.debug(`Version of the template is ${packageVersion}.`);
 
           return resolve({
             name: installedPkg.name,
-            version: installedPkg.version,
-            path: path.resolve(DEFAULT_TEMPLATES_DIR, installedPkg.name),
+            path: pkgPath
           });
         } catch (e) {
+          log.debug(`Errors while trying to resolve package location at ${pkgPath}`, e);
           // We did our best. Proceed with installation...
         }
       }
 
-      // NOTE: npmi creates symbolic links inside DEFAULT_TEMPLATES_DIR
-      // (node_modules) for local packages, i.e., those located in the file system.
-      npmi({
-        name: this.templateName,
-        install: force,
-        path: ROOT_DIR,
-        pkgName: 'dummy value so it does not force installation always',
-        npmLoad: {
-          loglevel: 'http',
-          save: false,
-          audit: false,
-          progress: false,
-          only: 'prod'
-        },
-      }, (err, result) => {
-        if (err) return reject(err);
-        resolve(beautifyNpmiResult(result));
+      const debugMessage = force ? 'as you passed --install flag' : 'because the template cannot be found on disk';
```

Comment:
Not saying "as" is incorrect there, just reads strangely to me:

```suggestion
      const debugMessage = force ? 'because you passed --install flag' : 'because the template cannot be found on disk';
```

Reply:
👍🏼 pushed fix here and in tests